workers/java/worker/src/main/java/com/hrznstudio/spatial/util/EntityRequirementCallback.java

```diff
@@ -0,0 +1,127 @@
+package com.hrznstudio.spatial.util;
+
+import improbable.Metadata;
+import improbable.MetadataData;
+import improbable.worker.ComponentMetaclass;
+import improbable.worker.Dispatcher;
+import improbable.worker.EntityId;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class EntityRequirementCallback {
+    //TODO: maybe make this delete old entities after a while to prevent high ram usage.
```

Comment:
You should subscribe to the entity deletion event and remove it from there, and also remove it once all required components have been set for that entity (and the callback called).